openstack/blockstorage/noauth/requests.go

```diff
@@ -0,0 +1,67 @@
+package noauth
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+// EndpointOpts specifies a "noauth" Cinder Endpoint.
+type EndpointOpts struct {
+	// CinderEndpoint [required] is currently only used with "noauth" Cinder.
+	// A cinder endpoint with "auth_strategy=noauth" is necessary, for example:
+	// http://example.com:8776/v2.
+	CinderEndpoint string
+}
+
+/*
+NewClient prepares an unauthenticated ProviderClient instance.
+Most users will probably prefer using the UnAuthenticatedClient function
+instead.
+*/
+func NewClient(options gophercloud.AuthOptions) (*gophercloud.ProviderClient, error) {
+	client := &gophercloud.ProviderClient{
+		TokenID: fmt.Sprintf("%s:%s", options.Username, options.TenantName),
+	}
+
+	return client, nil
+}
+
+// UnAuthenticatedClient allows for standalone "noauth" Cinder usage.
+func UnAuthenticatedClient(options gophercloud.AuthOptions) (*gophercloud.ProviderClient, error) {
```

Comment:
Can we remove this? It should be easy enough to pass in admin values to `NewClient`

Reply:
remove the entire "UnAuthenticatedClient" function?  are you saying just default to "admin" every time? 
As it is written today, someone can specify a user/tenant if they want to.

Reply:
Why do we need this function?

Reply:
Not to step on @jrperritt's feet, just dunno how busy he is to follow up :)

Since `UnAuthenticatedClient` and `NewClient` both take the same inputs and return the same values, the logic in `UnAuthenticatedClient` can easily be moved to `NewClient` like this:

```go
func NewClient(options gophercloud.AuthOptions) (*gophercloud.ProviderClient, error) {
	if options.Username == "" {
		options.Username = "admin"
	}
	if options.TenantName == "" {
		options.TenantName = "admin"
	}

	client := &gophercloud.ProviderClient{
		TokenID: fmt.Sprintf("%s:%s", options.Username, options.TenantName),
	}

	return client, nil
}
```

The returned `error` will always be `nil` with this combined function (technically it's always `nil` with `UnAuthenticatedClient` and `NewClient` anyway), but I think it's safe to leave -- maybe an actual error will be returned one day and the function signature won't have to be updated.

Reply:
well that's bloody ironic.

Reply:
works for me... was just trying to follow existing framework... agree this is cleaner.